### PR TITLE
Staging Sites: Clear state after sync and add success notification

### PR DESCRIPTION
**Pull request by kozer**
Related to https://github.com/Automattic/dotcom-forge/issues/4137

## Proposed Changes

This pr adds the following changes:

- Refactor code to avoid duplicate logic
- Add success notification when sync is completed
- Clear state for selected direction and sync options after a successful sync.

### Test 1: Clear options and toast notification
1. Create a WoA site. 
2. Add a staging site.
3. Initiate a sync.
4. Ensure that after the sync is completed, options are cleared and a toast with the message "'Site synced successfully." appears as shown:

2023-10-09T15:45:46,703172634+03:00
5. Ensure that when you initiate a sync again, the toast disappears.

### Test 2: Ensure that functionality is as before
Due to refactor, ensure that everything is working as described on https://github.com/Automattic/wp-calypso/pull/82477

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/hosting/staging-site-card/card-content/staging-sync-card.tsx [hunk 1]
```diff
@@ -325,164 +321,91 @@ const SyncCardContainer = ( {
...
+			setSelectedItems( [] );
+		}
+	}, [ isSyncInProgress ] );
 
-	const isSyncButtonDisabled =
-		disabled || ( selectedItems.length === 0 && selectedOption === 'pull' );
+	useEffect( () => {
+		if ( selectedOption && status === SiteSyncStatus.COMPLETED ) {
+			dispatch(
+				successNotice( translate( 'Site synced successfully.' ), { id: stagingSiteSyncSuccess } )
```
Comment (wojtekn) [inline client/my-sites/hosting/staging-site-card/card-content/staging-sync-card.tsx]:
```suggestion
				successNotice( translate( 'Site synchronized successfully.' ), { id: stagingSiteSyncSuccess } )
```

File: client/my-sites/hosting/staging-site-card/card-content/staging-sync-card.tsx [hunk 2]
```diff
@@ -510,7 +435,9 @@ export const ProductionSiteSyncCard = ( {
 				<FormLabel>
 					<FormRadio
 						className="staging-site-sync-card__radio"
-						label={ translate( 'Push to staging' ) }
+						label={ translate( 'Push to %s', {
+							args: [ type === 'production' ? 'staging' : 'production' ],
```
Comment (wojtekn) [inline client/my-sites/hosting/staging-site-card/card-content/staging-sync-card.tsx]: Is this approach translation-friendly?

Comment (kozer) [reply client/my-sites/hosting/staging-site-card/card-content/staging-sync-card.tsx]: Should I add the whole translate thing based on the type? Like so:

```
type === 'production ? translate('blah, blah') : translate('blom blom')
```

Comment (wojtekn) [reply client/my-sites/hosting/staging-site-card/card-content/staging-sync-card.tsx]: Yes, I believe this follows i18n best practices now.

**Review by wojtekn — APPROVED** [review]

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/9304372

Thank you @kozer for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/9304372).

**Pull request merged**